core/gpc_api.php

```diff
@@ -464,7 +464,13 @@ function gpc_make_array( $p_var_name ) {
  * @return boolean
  */
 function gpc_string_to_bool( $p_string ) {
-	if( 0 == strcasecmp( 'off', $p_string ) || 0 == strcasecmp( 'no', $p_string ) || 0 == strcasecmp( 'false', $p_string ) || 0 == strcasecmp( '', $p_string ) || 0 == strcasecmp( '0', $p_string ) ) {
+	if( 0 == strcasecmp( 'off', $p_string )
+		|| 0 == strcasecmp( 'no', $p_string )
+		|| 0 == strcasecmp( 'n', $p_string )
+		|| 0 == strcasecmp( 'false', $p_string )
+		|| 0 == strcasecmp( '', $p_string )
+		|| 0 == strcasecmp( '0', $p_string )
```

Comment:
One more thing... `strcasecmp` call is not needed for the last 2 items. Furthermore, to avoid multiple function calls, I would suggest to start by converting $p_string to lowercase; you can then perform simple string comparisons for all values.

Finally, wouldn't it be better to test against "truthy" values and default anything else to false, instead of the reverse as you've currently implemented it ?

Reply:
That function has not been changed, apart from adding the new option for parsing "y/n"

>  I would suggest to start by converting $p_string to lowercase

ok

> Wouldn't it be better to test against "truthy" values and default anything else to false, instead of the reverse

That makes sense. However note that it should be refactored with care. As i explained, adding the "y/n" case was needed because we already used "temporary=y" for a long time (see congif_defauls_inc), which defaulted to true even if it was not a supported boolean string. Other similar cases may exist in the wild.

Reply:
In that case, let's keep it as it is